src/pipeline/src/manager/table.rs

```diff
@@ -0,0 +1,450 @@
+}
+
+impl PipelineTable {
+    /// Create a new PipelineTable.
+    pub fn new(
+        inserter: InserterRef,
+        statement_executor: StatementExecutorRef,
+        table: TableRef,
+        query_engine: QueryEngineRef,
+    ) -> Self {
+        Self {
+            inserter,
+            statement_executor,
+            table,
+            query_engine,
+            pipelines: Cache::builder()
+                .max_capacity(PIPELINES_CACHE_SIZE)
+                .time_to_live(Duration::from_secs(10))
+                .build(),
+        }
+    }
+
+    /// Build the schema for the pipeline table.
+    pub fn build_pipeline_schema() -> (String, Vec<String>, Vec<ColumnDef>) {
+        (
+            PIPELINE_TABLE_CREATED_AT_COLUMN_NAME.to_string(),
+            vec![
+                PIPELINE_TABLE_PIPELINE_SCHEMA_COLUMN_NAME.to_string(),
+                PIPELINE_TABLE_PIPELINE_NAME_COLUMN_NAME.to_string(),
+                PIPELINE_TABLE_PIPELINE_CONTENT_TYPE_COLUMN_NAME.to_string(),
+            ],
+            vec![
+                ColumnDef {
+                    name: PIPELINE_TABLE_PIPELINE_NAME_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::String as i32,
+                    is_nullable: false,
+                    default_constraint: vec![],
+                    semantic_type: SemanticType::Tag as i32,
+                    comment: "".to_string(),
+                    datatype_extension: None,
+                },
+                ColumnDef {
+                    name: PIPELINE_TABLE_PIPELINE_SCHEMA_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::String as i32,
+                    is_nullable: false,
+                    default_constraint: vec![],
+                    semantic_type: SemanticType::Tag as i32,
+                    comment: "".to_string(),
+                    datatype_extension: None,
+                },
+                ColumnDef {
+                    name: PIPELINE_TABLE_PIPELINE_CONTENT_TYPE_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::String as i32,
+                    is_nullable: false,
+                    default_constraint: vec![],
+                    semantic_type: SemanticType::Tag as i32,
+                    comment: "".to_string(),
+                    datatype_extension: None,
+                },
+                ColumnDef {
+                    name: PIPELINE_TABLE_PIPELINE_CONTENT_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::String as i32,
+                    is_nullable: false,
+                    default_constraint: vec![],
+                    semantic_type: SemanticType::Field as i32,
+                    comment: "".to_string(),
+                    datatype_extension: None,
+                },
+                ColumnDef {
+                    name: PIPELINE_TABLE_CREATED_AT_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::TimestampNanosecond as i32,
```

Comment:
I think `millisecond` is enough as the time range that nanoseconds can represent is quite narrow.